Sources/MapboxNavigation/DayStyle.swift

```diff
@@ -291,6 +291,9 @@ open class DayStyle: Style {
         WayNameView.appearance().backgroundColor = UIColor.defaultRouteLayer.withAlphaComponent(0.85)
         WayNameView.appearance().borderColor = UIColor.defaultRouteCasing.withAlphaComponent(0.8)
         WayNameView.appearance().borderWidth = 1.0
+        StepsTableHeaderView.appearance().normalTextColor = #colorLiteral(red: 0.431372549, green: 0.431372549, blue: 0.431372549, alpha: 1)
+        StepsTableHeaderView.appearance().normalBackgroundColor = #colorLiteral(red: 1, green: 1, blue: 1, alpha: 1)
+        StepsTableHeaderView.appearance().normalFont = UIFont.systemFont(ofSize: 18, weight: .medium).adjustedFont
```

Comment:
This styling doesn't seem to match what we had before. For example:

Current state:
![Simulator Screen Shot - iPhone SE (2nd generation) - 2022-02-23 at 12 40 46](https://user-images.githubusercontent.com/1496498/155405564-c2b5ef74-1869-41ff-bc24-234cedc905c5.png)

Previous state:
![Simulator Screen Shot - iPhone SE (2nd generation) - 2022-02-23 at 12 44 28](https://user-images.githubusercontent.com/1496498/155405609-a1ea2c84-1ccd-48af-8b73-aa88c221a294.png)

There is also a problem when additional empty spacing is being added:
![Simulator Screen Shot - iPhone SE (2nd generation) - 2022-02-23 at 12 38 14](https://user-images.githubusercontent.com/1496498/155405725-c5950f6c-807f-4758-943c-667109d58d17.png)

Reply:
The spacing difference is caused by the introduction of the `UITableViewHeaderFooterView`. If we didn't make the change, before ios15.0, we had narrow spacing of the section header, and the white section header at night style. While after iOS15.0, the default section header is wider, which is same as the current implementation of the `UITableViewHeaderFooterView`. And also the iOS would have the default same color as the table cell in iOS15.0, so we didn't make the change here. 
So this PR is to make the mobile have the same UI appearance in iOS14 and after 15.

Reply:
@MaximAlien, the extra padding effect is brought by the default setting of iOS15, which is removed. And the highlighted effect introduced from iOS 15 is also disabled. So right now the iOS version before and after 15 have the same UI appearance now with the new fix.